### PR TITLE
Add python subprocess32 package

### PR DESCRIPTION
**Pull request by bobhenz-jabil**
https://packages.debian.org/stretch/python-subprocess32
https://apps.fedoraproject.org/packages/python-subprocess32
https://packages.gentoo.org/packages/dev-python/subprocess32
https://packages.ubuntu.com/bionic/python-subprocess32

### CODE REVIEW

File: rosdep/python.yaml [hunk 1]
```diff
@@ -3647,6 +3647,11 @@ python-sqlalchemy:
     wily: [python-sqlalchemy]
     xenial: [python-sqlalchemy]
     xenial_python3: [python3-sqlalchemy]
+python-subprocess32:
+  debian: [python-subprocess32]
+  fedora: [python-subprocess32]
+  gentoo: [dev-python/subprocess32]
+  ubuntu: [python-subprocess32]
```
Comment (mikaelarguedas) [inline rosdep/python.yaml]: This package is not available on trusty, so we should explicitely set the rule to `null` for that platform

File: rosdep/python.yaml [hunk 2]
```diff
@@ -3647,6 +3647,11 @@ python-sqlalchemy:
     wily: [python-sqlalchemy]
     xenial: [python-sqlalchemy]
     xenial_python3: [python3-sqlalchemy]
+python-subprocess32:
+  debian: [python-subprocess32]
```
Comment (mikaelarguedas) [inline rosdep/python.yaml]: This package is not available on jessie, so we should explicitely set the rule to `null` for that platform

**Review by mikaelarguedas — CHANGES_REQUESTED** [review]
@bobhenz-jabil thanks for contributing. Can you please provide a description of the package as well a the use cases you need it for as per [contribution guildelines](https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#rosdep-rules-contributions).

**Comment by bobhenz-jabil** [comment]
Brief Description: subprocess32 is a backport of the Py3 stdlib subprocess module for Py2. 

Use case: One of the very valuable features of this package is the ability to add a "timeout" parameter to various subprocess methods preventing calls to the system from hanging nodes indefinitely.

**Review by mikaelarguedas — APPROVED** [review]
lgtm thanks @bobhenz-jabil for iterating

**Pull request merged**